src/actions/actions.ts

```diff
@@ -1723,12 +1710,67 @@ function searchCurrentWord(position: Position, vimState: VimState, direction: Se
       ? vimState.cursorPosition.getWordLeft(true)
       : vimState.cursorPosition;
 
-    vimState.cursorPosition = vimState.globalState.searchState.getNextSearchMatchPosition(searchStartCursorPosition).pos;
+    return performSearchMovement({
+      needle: currentWord,
+      vimState,
+      direction,
+      isExact,
+      searchStartCursorPosition
+    });
+}
+
+function searchCurrentSelection (vimState: VimState, direction: SearchDirection, isExact: boolean) {
+    const selection = TextEditor.getSelection();
+    const end = new Position(selection.end.line, selection.end.character + 1);
+    const currentSelection = TextEditor.getText(selection.with(selection.start, end));
+
+    // Go back to Normal mode, otherwise the selection grows to the next match.
+    vimState.currentMode = ModeName.Normal;
+
+    // If the search is going left then use `getLeft()` on the selection start.
+    // If going right then use `getRight()` on the selection end. This ensures
+    // that any matches happen outside of the currently selected word.
+    const searchStartCursorPosition = direction === SearchDirection.Backward
+      ? vimState.lastVisualSelectionStart.getLeft()
+      : vimState.lastVisualSelectionEnd.getRight();
+
+    return performSearchMovement({
+      needle: currentSelection,
+      vimState,
+      direction,
+      isExact,
+      searchStartCursorPosition
+    });
+}
+
+interface IPerformSearchMovementArgs {
```

Comment:
Yeah I asked about this on slack, because the inlined arguments made the linter complain about the line length. So the question was numerous arguments over a new line versus an interface.